src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_NUMBER = 20;
+    private static final int FIGURES_NUMBER = 3;
+    private static final int DEFAULT_RADIUS_CIRCLE = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FIGURES_NUMBER);
+        String color = colorSupplier.getRandomColor();
+        switch (index) {
+            case 0:
+                int shortSide = random.nextInt(MAX_RANDOM_NUMBER);
+                int longSide = random.nextInt(MAX_RANDOM_NUMBER);
+                return new Rectangle(color, shortSide, longSide);
+            case 1:
+                int side = random.nextInt(MAX_RANDOM_NUMBER);
+                return new Square(color, side);
+            case 2:
+                int firstLeg = random.nextInt(MAX_RANDOM_NUMBER);
+                int secondLeg = random.nextInt(MAX_RANDOM_NUMBER);
+                return new RightTriangle(color, firstLeg, secondLeg);
+            default:
+                return new Square(colorSupplier.getRandomColor(),random.nextInt(MAX_RANDOM_NUMBER));
```

Comment:
you create array with 5 figures just to pick one of them. Lets use switch-case construction. We can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction

Reply:
sorry, could you please explain?... i didnt understand what you mean... because i create random ranctangle, square and  RightTriangle in that switch-case

Reply:
sorry, wrong comment) Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction.
```
        switch (index) {
            case 0:
                return getRandomRectangle();
            case 1:
                return getRandomSquare();
                ....
```